src/gui/DatabaseTabWidget.cpp

```diff
@@ -129,7 +129,12 @@ void DatabaseTabWidget::openDatabase(const QString& fileName, const QString& pw,
     while (i.hasNext()) {
         i.next();
         if (i.value().canonicalFilePath == canonicalFilePath) {
-            setCurrentIndex(databaseIndex(i.key()));
+            if (!i.value().dbWidget->dbHasKey() && (!pw.isNull() || !keyFile.isEmpty())) {
```

Comment:
I would prefer the logically equivalent `!(pw.isNull() && keyFile.isEmpty())`, which is more intuitive to understand.

Reply:
This logic sequence is used 15 lines below as well.

Reply:
Change it there, too. ;-)